examples/seismic/skew_self_adjoint/operators.py

```diff
@@ -0,0 +1,330 @@
+
+    # Create interpolation expression for receivers, extracting at dp(t)
+    rec_term = rec.interpolate(expr=du)
+
+    # Substitute spacing terms to reduce flops
+    dt = time_axis.step
+    spacing_map = v.grid.spacing_map
+    spacing_map.update({t.spacing: dt})
+
+    return Operator(eqn1 + src_term + eqn2 + rec_term, subs=spacing_map,
+                    name='ISO_JacobianFwdOperator', **kwargs)
+
+
+def ISO_JacobianAdjOperator(model, rec, time_axis, space_order=8,
+                            save=True, **kwargs):
+    """
+    Construct a linearized JacobianAdjoint modeling operator in a variable density
+    visco- acoustic media.
+
+    Parameters
+    ----------
+    model : Dictionary <string>:<Function>, contains:
+        'b': Buoyancy = reciprocal density (units: m^3/kg)
+        'v': Velocity (units: m/msec or km/sec)
+        'wOverQ': The w/Q field for dissipation only attenuation.
+    rec : SparseTimeFunction (PointSource)
+        Receiver positions and time siganture.
+    time_axis : TimeAxis
+        Defines temporal sampling
+    space_order : int, optional, Defaults to 8
+        Space discretization order.
+    save : int or Buffer, optional
+        Saving flag, True saves all time steps. False saves three timesteps.
+        Defaults to False.
+
+    Returns
+    ----------
+    The operator implementing Jacobian adjoint modeling
+    """
+    # Get the Functions for buoyancy, velocity, wOverQ
+    assert 'b' in model, "model dictionary must contain key 'b'"
+    assert 'v' in model, "model dictionary must contain key 'v'"
+    assert 'wOverQ' in model, "model dictionary must contain key 'wOverQ'"
+    b = model['b']
+    v = model['v']
+    wOverQ = model['wOverQ']
+
+    # Create p0, dp wavefields and dm velocity perturbation field
+    u0 = TimeFunction(name="u0", grid=v.grid,
+                      save=time_axis.num if save else None,
+                      time_order=2, space_order=space_order)
+
+    du = TimeFunction(name="du", grid=v.grid,
+                      save=time_axis.num if save else None,
+                      time_order=2, space_order=space_order)
+
+    dm = Function(name="dm", grid=v.grid, space_order=space_order)
+
+    # Time update equation
+    t = u0.dimensions[0]
+    eqn = iso_stencil(du, model, forward=False)
+    dm_update = Eq(dm, dm +
+                   du * (2 * b * v**-3 * (wOverQ * u0.dt(x0=t-t.spacing/2) + u0.dt2)))
+
+    # Construct expression to inject receiver values, injecting at p(t-dt)
+    rec_term = rec.inject(field=du.backward, expr=rec * t.spacing**2 * v**2 / b)
+
+    # Substitute spacing terms to reduce flops
+    dt = time_axis.step
+    spacing_map = v.grid.spacing_map
+    spacing_map.update({t.spacing: dt})
+
+    return Operator([dm_update] + eqn + rec_term, subs=spacing_map,
+                    name='ISO_JacobianAdjOperator', **kwargs)
+
+    # note this is broken somehow: possibly a different path is chosen through the AST?
```

Comment:
we really need an MFE for this and an issue on github

Reply:
ok if I leave that comment for now?